### ISSUE TITLE
[Bug]: `screen.getAllDisplays()` and `desktopCapturer.getSources()` order no longer matches (29.3.2, 30.0.2)

### ISSUE THREAD

**Issue by Galkon** [electron/electron#42031]
### Preflight Checklist

- [X] I have read the [Contributing Guidelines](https://github.com/electron/electron/blob/main/CONTRIBUTING.md) for this project.
- [X] I agree to follow the [Code of Conduct](https://github.com/electron/electron/blob/main/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] I have searched the [issue tracker](https://www.github.com/electron/electron/issues) for a bug report that matches the one I want to file, without success.

### Electron Version

29.3.2, 30.0.2

### What operating system are you using?

macOS

### Operating System Version

Sonoma 14.4.1

### What arch are you using?

arm64 (including Apple Silicon)

### Last Known Working Electron version

28.3.1

### Expected Behavior

When using multiple displays, it is expected that `screen.getAllDisplays()` returns the displays in the same order as sources returned from:
```
await desktopCapturer.getSources({
    types: ['screen'],
    thumbnailSize: thumbnailSize
  })
```

- Display index: 0
  - Screen display ID by index: 3
  - Source display ID by index: 3
- Display index: 1
  - Screen display ID by index: 1
  - Source display ID by index: 1
- Display index: 2
  - Screen display ID by index: 2
  - Source display ID by index: 2

https://github.com/electron/electron/assets/1406556/af473042-ba26-4fc8-8363-e77eb546fd22

### Actual Behavior

In 29.3.2, 30.0.1, 30.0.2, on my setup with 3 monitors, it started providing mismatching orders between the two every time.

- Display index: 0
  - Screen display ID by index: 3
  - Source display ID by index: 2
- Display index: 1
  - Screen display ID by index: 1
  - Source display ID by index: 3
- Display index: 2
  - Screen display ID by index: 2
  - Source display ID by index: 1

https://github.com/electron/electron/assets/1406556/6cc358dc-cec5-42b3-b974-f05c13d80f92

### Testcase Gist URL
https://gist.github.com/Galkon/f467fdaf8f03cc0aac9f811e2a54c04d

### Additional Information

This _was_ working for me on this computer earlier this week. I have a KVM switch, which I had to rearrange the ports yesterday, and since then I've had this issue.

**Downgrading to electron 28.3.1 resolves the issue. Upgrading back to 29.3.2 or 30.0.1, 30.0.2 again reintroduces the issue.**

I've since changed my code to match `display.id` to `sources.display_id` to get around this unexpected behavior -- obviously this change will be more reliable -- but I wasn't sure if it was intended for this to have changed or not.

**Comment by Galkon** [comment]
Working on supplying a gist for this -- it looks like it actually may not be with the cursor coordinates but with the index from this:
```
  const displays = screen.getAllDisplays()
  const displayIndex = displays.findIndex((disp) => disp.id === activeDisplay.id)
```

No longer matching the order of the sources from this:
```
  const sources = await desktopCapturer.getSources({
    types: ['screen'],
    thumbnailSize: thumbnailSize
  })
  const source = sources[displayIndex]
```

Will update issue accordingly with complete gist when finished debugging.

**Comment by georgexu99** [comment]
https://github.com/electron/electron/pull/42781 and https://github.com/electron/electron/pull/42883/ fix the same issue on windows

**Comment by georgexu99** [comment]
Ah, Unfortunately it seems to be still a problem. My apologies for the mistake!

**Comment by georgexu99** [comment]
Hey! I'm coming back with an explanation for the behaviour but TLDR:
`screen.getAllDisplays()` and `desktopCapturer.getSources()` don't guarantee an order to the list of desktop ids returned, so I'll be closing this bug as it is intended behaviour. 

I managed to bisect the order change to electron [29.0.1 to 29.1.0](https://github.com/electron/electron/compare/v29.0.1...v29.1.0), and specifically, commit [fix: use ScreenCaptureKit exclusively on macOS 14.4 and higher](https://github.com/electron/electron/commit/ce2ac1c0c24171b8b1b4bcf5a3345e7bb80474d4). This is when we migrated to Apple's at the time, new [ScreenCaptureKit API](https://developer.apple.com/documentation/screencapturekit/scshareablecontent) from [CGDisplay](https://developer.apple.com/documentation/coregraphics/1454964-cggetonlinedisplaylist). Thus the ordering is different due to the underlying API differences!
